dashboard/lib/main.dart

```diff
@@ -91,26 +96,68 @@ class MyApp extends StatelessWidget {
           ),
         ),
         darkTheme: ThemeData.dark(),
+
+        // The default page is the build dashboard.
         initialRoute: BuildDashboardPage.routeName,
-        routes: <String, WidgetBuilder>{
-          BuildDashboardPage.routeName:
-              (BuildContext context) => const BuildDashboardPage(),
+        routes: {
+          BuildDashboardPage.routeName: (_) => const BuildDashboardPage(),
         },
+
+        // dashboard.com/x/flutter/flutter/presubmit
+        // jonahwilliams@ [1234] > .../1234
+        // matanlurey@    [4568] > .../4567
+        //
+        // dashboard.com/x/flutter/flutter/presubmit/1234
         onGenerateRoute: (RouteSettings settings) {
           final uriData = Uri.parse(settings.name!);
-          if (uriData.path == BuildDashboardPage.routeName) {
-            return MaterialPageRoute<void>(
-              settings: RouteSettings(name: uriData.toString()),
-              builder: (BuildContext context) {
-                return BuildDashboardPage(
-                  queryParameters: uriData.queryParameters,
+          if (uriData.pathSegments.isEmpty) {
+            return null;
+          }
+
+          switch (uriData.pathSegments.first) {
+            case BuildDashboardPage.routeName:
+              return MaterialPageRoute<void>(
+                settings: settings,
+                builder: (BuildContext context) {
+                  return BuildDashboardPage(
+                    queryParameters: uriData.queryParameters,
+                  );
+                },
+              );
+            case 'v2':
+              if (_findV2Route(uriData) case final builder?) {
+                return MaterialPageRoute<void>(
+                  builder: builder,
+                  settings: settings,
                 );
-              },
-            );
+              }
           }
           return null;
         },
       ),
     );
   }
+
+  WidgetBuilder? _findV2Route(Uri route) {
+    if (route.pathSegments.isEmpty || route.pathSegments.first != 'v2') {
+      throw ArgumentError.value(route, 'route', 'not a v2 route');
+    }
+
+    final [_, ...v2PathSegments] = route.pathSegments;
+    if (v2PathSegments.isEmpty) {
+      return (_) => const V2LandingPage();
+    }
+
+    final [repoOwner, ...] = v2PathSegments;
+    if (v2PathSegments.length == 1) {
+      return (_) => V2LandingPage(repoOwner);
+    }
+
+    final [_, repoName, ...] = v2PathSegments;
+    if (v2PathSegments.length == 2) {
+      return (_) => V2LandingPage(repoOwner, repoName);
+    }
+
```

Comment:
This got me hacking; not a nit or anything else:

```dart
   switch (v2PathSegments) {
     case [var repoOwner, var repoName, ...]:
         print('repoOwner/Name: $repoOwner $repoName');
     case [var repoOwner, ...]:
         print('repoOwner: $repoOwner');
     default:
         print('empty');
   }
```

Without the whole switch/case:

```dart
  if (v2PathSegments case [var repoOwner, var repoName, ...]) {
    print('repoOwner/Name: $repoOwner $repoName');
    return;
  }
  if (v2PathSegments case [var repoOwner, ...]) {
      print('repoOwner: $repoOwner');
    return;
  }  
  print('empty');
  ```

Reply:
Done.